helios-client/src/main/java/com/spotify/helios/client/HeliosClient.java

```diff
@@ -615,8 +601,30 @@ private HttpConnector createHttpConnector(final boolean sslHostnameVerification)
         log.debug("{}", e);
       }
 
-      return new AuthenticatingHttpConnector(user, agentProxyOpt, clientCertificatePath,
-                                             clientKeyPath, endpointIterator, connector);
+      // set up the ClientCertificatePath, giving precedence to any values set
+      // with setClientCertificatePath()
+      ClientCertificatePath clientCertificatePath = this.clientCertificatePath;
+      if (clientCertificatePath == null) {
+        final String heliosCertPath = System.getenv(HELIOS_CERT_PATH);
+        if (!isNullOrEmpty(heliosCertPath)) {
+          final Path certPath = Paths.get(heliosCertPath, "cert.pem");
+          final Path keyPath = Paths.get(heliosCertPath, "key.pem");
+
+          if (certPath.toFile().canRead() && keyPath.toFile().canRead()) {
```

Comment:
Is there any benefit to using a try catch here and letting the `ClientCertificatePath` constructor handle the logic?